rust-bindings/src/lib.rs

```diff
@@ -0,0 +1,194 @@
+#![allow(non_camel_case_types)]
+#![allow(non_snake_case)]
+#![allow(dead_code)]
+#![allow(non_upper_case_globals)]
+
+use std::fmt;
+
+mod bindings {
+    include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
+}
+
+/// The encoded form of a BIP 158 filter.
+pub struct EncodedFilter(bindings::Slice);
+
+impl fmt::Debug for EncodedFilter {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.as_ref().fmt(f)
+    }
+}
+
+impl PartialEq for EncodedFilter {
+    fn eq(&self, other: &Self) -> bool {
+        self.as_ref() == other.as_ref()
+    }
+}
+
+impl Eq for EncodedFilter {}
+
+impl AsRef<[u8]> for EncodedFilter {
+    fn as_ref(&self) -> &[u8] {
+        // SAFETY: The `Slice` struct is guaranteed to be valid for the lifetime of the `EncodedFilter` struct.
+        unsafe { std::slice::from_raw_parts(self.0.bytes, self.0.length) }
+    }
+}
+
+impl Drop for EncodedFilter {
+    fn drop(&mut self) {
+        // SAFETY: The `Slice` struct is guaranteed to be valid for the lifetime of the `EncodedFilter` struct.
+        unsafe { bindings::free_slice(self.0) }
+    }
+}
+
+/// A BIP 158 filter.
+pub struct Bip158Filter(*const bindings::GCSFilter);
+
+impl Bip158Filter {
+    /// Encode a BIP 158 filter from a list of slices.
+    pub fn new(slices: &[&[u8]]) -> Self {
```

Comment:
it's a bit of a nit-pick, but it would be a bit flexible to take a `&[Borrow<[u8]>]`. Then you could pass in both `&[Vec<u8>]` and `&[&[u8]]`